org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalRequestor.java

```diff
@@ -120,7 +120,8 @@ public int compare(CompletionProposal p1, CompletionProposal p2) {
 				}
 				res = p2Length - p1Length;
 			}
-			if (res == 0) {
+			if (res == 0 && (p1.getKind() == CompletionProposal.METHOD_REF || p1.getKind() == CompletionProposal.CONSTRUCTOR_INVOCATION)
```

Comment:
I don't think it's as simple as method reference and constructor invocation. See 

https://github.com/eclipse-jdtls/eclipse.jdt.ls/blob/07f65f35dde668960f34c81af03a49d461609670/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalRequestor.java#L523-L529

https://github.com/eclipse-jdtls/eclipse.jdt.ls/blob/07f65f35dde668960f34c81af03a49d461609670/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalRequestor.java#L486-L488

There's other types of completion proposal kinds that we'd have to make sure don't trigger this. `mapKind` seems like the right method to use. You'd need to make it static, but I think that should be ok.